docs/development/release.md

```diff
@@ -3,148 +3,297 @@
+
+1. Update your local Helm repository:
+
+    ```bash
+    helm repo add kuberay https://ray-project.github.io/kuberay-helm/
+    helm repo update kuberay
+    ```
+
+2. Search for the new chart versions:
+
+    ```bash
+    helm search repo kuberay/kuberay-operator --versions
+    helm search repo kuberay/ray-cluster --versions
+    # Verify that version 1.4.0 (or your target version) appears
+    ```
+
+3. Install KubeRay using the new chart versions on a test cluster (e.g., Kind, Minikube):
+
+    ```bash
+    # Example using Kind
+    kind create cluster
+
+    # Install the operator using the specific version
+    helm install kuberay-operator kuberay/kuberay-operator --version <new-version>
+
+    # Install a sample RayCluster using the specific chart version
+    helm install raycluster kuberay/ray-cluster --version <new-version>
+    ```
+
+4. Check the pods and basic functionality:
+
+    ```bash
+    kubectl get pods
+    ```
 
-#### Step 7. Update KubeRay documentation in Ray repository
+    You should see the kuberay-operator pod in `Running` state and 1 active RayCluster.
 
-* Update KubeRay documentation in Ray repository with v0.5.0. Examples for v0.5.0:
-  * <https://github.com/ray-project/ray/pull/33339>
-  * <https://github.com/ray-project/ray/pull/34178>
+---
 
-#### Step 8. Generate release
+### Step 8: Generate the CHANGELOG
 
-* Manually trigger the `release-kubectl-plugin` CI job to generate the release.
-* Follow the [instructions](../release/changelog.md) to generate release notes and add notes in the GitHub release.
+Follow [Generating the changelog for a release](https://github.com/ray-project/kuberay/blob/master/docs/release/changelog.md) to generate the change log for the new release.
 
-#### Step 9. Announce the release on the KubeRay slack
+---
 
-* Announce the release on the KubeRay slack ([example](https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1681244150758839))!
+### Step 9: Publish Release Notes
 
-#### Step 10. Update CHANGELOG
+1. Go to the [KubeRay Releases page](https://github.com/ray-project/kuberay/releases).
+2. Find the **draft release** that was created automatically in Step 5 or create a new one if needed, targeting the tag `v1.4.0`.
+3. Click **"Edit"** on the draft release.
+4. Paste the generated **CHANGELOG** content from Step 8 into the release description.
+5. Add any additional release highlights, bug fixes, known issues, etc.
+6. Ensure the correct tag (`v1.4.0`) is selected.
+7. Verify the attached assets (kubectl plugins) are correct.
+8. Mark the release as **"Latest release"** if appropriate (usually for the newest stable release).
+9. Click **"Publish release"**.
 
-* Send a PR to add the release notes to [CHANGELOG.md](../../CHANGELOG.md).
+---
 
-#### Step 11. Update and improve this release document
+### Step 10: Update ray.io documentation
 
-* Update this document and optimize the release process!
+Update all references to the KubeRay version in the ray.io documentation.
```

Comment:
Sometimes we also need to update the `rayVersion` and the image tag for Ray image in this folder https://github.com/ray-project/kuberay/tree/master/ray-operator/config/samples.

And after my doctest PR (it has been merged), we also need to change the ray version in https://github.com/ray-project/ray/blob/30454820bda11eb740cd8335121797d4de71633d/ci/k8s/run-kuberay-doc-tests.sh. And the kuberay version in https://github.com/ray-project/ray/blob/30454820bda11eb740cd8335121797d4de71633d/doc/source/cluster/kubernetes/scripts/doctest-utils.sh.

Also, how about add a step to say that "Ensure all doctests passed after updating the doc"?